HIP/hip-1056.md

```diff
@@ -0,0 +1,1769 @@
+
+A subsequent HIP will be created to outline the changes and justification required for implementing Hedera TSS-BLS
+signatures.
+
+#### Design for State Changes
+
+Including state changes within the block stream is a significant foundational step towards enhancing the
+decentralization of the network. This inclusion allows for the development of alternative node implementations and
+enables downstream state-related services such as state proofs, state snapshots, and reconnect services. It also
+provides a low latency feed of the contents of state as it changes to applications. This can open doors to exciting new
+latency critical application ideas and concepts that could not be built before.
+
+To enable this, state in the consensus node will be restructured into simple named states. This allows for state changes
+to be expressed as simple CRUD operations on named states. This is so downstream consumers do not have to understand the
+full complexity of the merkle state tree.
+
+## Personas and User Stories
+
+**Personas:**
+- _Downstream Developer_: A provider of value-added services utilizing block streams
+- _Verifier_: Any person or service required to verify the integrity of a block or the information contained within a
+  block.
+
+**User Stories:**
+1. As a downstream developer, I want a streamlined feed that contains all event, transactional, and state details of
+   the network so that I can offer value-added services to the ecosystem.
+2. As a downstream developer, I want each block of data to be verifiable with a single, aggregated signature, so that I
+   can authenticate transactional data without the cost and complexity of downloading multiple signature files for
+   verification.
+3. As a verifier for a network, I want a single, self-contained stream of data with aggregated signature signed by a
+   threshold of stake weights, so that I can cost-effectively confirm that the stream represents the consensus output of
+   the network.
+4. As a downstream developer, I want a low latency feed that provides the state changes associated with transaction(s)
+   within a block, so that I can interact with state changes with the smallest possible delay.
+
+## Specification
+
+The Block Stream will be defined by the continuous transmission of Block information defined as follows.
+- Each `Block` represents all inputs and outputs of the consensus network for zero or more whole `Rounds`. The maximum
+  number of rounds per block will be configurable and start at 1 round per block. It could be increased later if
+  necessary for performance reasons or if round become very frequent and small.
+    - A block *may* be zero rounds because the last block prior to a network "freeze" must communicate the state hash
+      (via a block proof) of the state immediately prior to network shutdown. This is only possible with an extra empty
```

Comment:
I'm confused by the `(via a block proof)` part here. In normal cases, the state hash is part of the block merkle tree and the block proof has an aggregated signature of the block merkle tree root (although now I realize that this has not been explicitly state in the HIP thus far, and it probably should be). Is this section saying that in a block with no rounds, the proof is simply on the hash of the state root and not a merkle tree root hash that also includes previous block root hash and null padding?

Reply:
A block with no rounds is the same as any other block except for the lack of any content between block header and block proof.
The bullet point is describing a specific case where a block with no rounds may
be required, and describing _how_ that "empty" block resolves a sequence problem.

The wording is _intended_ to explain that, because block proof contains the state hash at the beginning of the block (end of the prior block), the state hash at end-of-block is communicated in the block stream as a field in the block proof for the following block.

Reply:
>the state hash at end-of-block is communicated in the block stream as a field in the block proof

Aha! I had not gotten to the BlockProof proto yet.